code/datums/gamemode/objectives/target/assassinate.dm

```diff
@@ -44,8 +53,47 @@
 		return TRUE
 	//Borgs/brains/AIs count as dead for traitor objectives. --NeoFite
 	// If they're in an away mission/custom z-level by the time you're checking this might as well count them as MIA
+	switch (syndicate_checked)
+		if (SYNDICATE_VALIDATED)
+			return TRUE
+		if (SYNDICATE_CANCELED)
+			return FALSE
 	if(target && target.current)
 		if(target.current.stat == DEAD || issilicon(target.current) || isbrain(target.current) || target.current.z > 6 || !target.current.ckey || isborer(target.current))
 			return TRUE
 		return FALSE
-	return TRUE
\ No newline at end of file
+	return TRUE
+
+/datum/objective/target/assassinate/proc/SyndicateCertification()
+	if (syndicate_checked == SYNDICATE_CANCELED)//if your death was reported prior, you're out of the race. Better luck next time.
+		return
+
+	syndicate_checked = SYNDICATE_VALIDATED
+
+	//The syndicate has confirmed that the double agent has taken out their target.
+	//They will now assign the new objective of assassinating their old target's target.
+	//Unless said target is themselves, which then means that all other agents have been eliminated and they have won.
+	var/datum/role/traitor/challenger/self = owner.GetRole(CHALLENGER)
+	var/datum/role/traitor/challenger/enemy = target.GetRole(CHALLENGER)
+	if (!self ||!enemy)
+		message_admins("a double agent syndicate certification has failed. call deity.")//not that this should ever happen
+		return
+
+	for (var/datum/objective/target/assassinate/A in enemy.objectives.objectives)
+		if (A.syndicate_checked)
+			continue
+		if (A.target == owner)
```

Comment:
This looks like a rather convoluted way to check if there are no other agents left. At minimum, put a comment here explaining that this condition means the guy is the last in the race.

Reply:
> At minimum, put a comment here explaining that this condition means the guy is the last in the race.

But I did, literally 10 lines above.

Reply:
Seems fine to me.